Lazydocs: API Reference doc refresh (#1424)
wandb/docs: content/en/guides/models/track/limits.md

## Logging considerations

> Use `run.log` to track experiment metrics. Once logged, these metrics generate charts and show up in tables. Too much logged data can make the app slow.

Reviewer: ```suggestion
Use `run.log` to track experiment metrics. Logged metrics generate charts and show up in tables. Too much logged data can make the app slow.
```
"shows up in tables" seems weird. Do we mean Table panels? Is this whole thing talking about automatic workspaces getting slow?

> Pick a logging frequency that is appropriate to the metric you are logging. As a general rule of thumb, the wider the metric the less frequently you should log it. W&B recommends:

Reviewer: ```suggestion
Pick a logging frequency that is appropriate to the metric you are logging. As a general rule of thumb, log wider values less frequently than narrower values. W&B recommends:
```

- Scalars: <100,000 logged points per metric
- Media: <50,000 logged points per metric
- Histograms: <10,000 logged points per metric

```python
# Training loop with 1m total steps
for step in range(1000000):
    #  not recommended
    run.log(
        {
            "scalar": step,  # 100,000 scalars
            "media": wandb.Image(...),  # 100,000 images
            "histogram": wandb.Histogram(...),  # 100,000 histograms
        }
    )

    #  recommended
    if step % 1000 == 0:
        run.log(
            {
                "histogram": wandb.Histogram(...),  # 10,000 histograms
            },
            commit=False,
        )
    if step % 200 == 0:
        run.log(
            {
                "media": wandb.Image(...),  # 50,000 images
            },
            commit=False,
        )
    if step % 100 == 0:
        run.log(
            {
                "scalar": step,  # 100,000 scalars
            },
            commit=True,
        )  # Commit batched, per-step metrics together
```

> <!-- Enable batching in calls to `run.log` by passing `commit=False` to minimize the total number of API calls for a given step. See [the docs]({{< relref "/ref/python/sdk/classes/run/#method-runlog" >}}) for `run.log` for more details. -->

Reviewer: Intentionally commented out?

### Metric count

> When logging between 5000 and 100,000 metrics per run, W&B recommends using a [manual workspace]({{< relref "/guides/models/app/features/panels/#workspace-modes" >}}). In Manual mode, you can easily add and remove panels in bulk as you choose to explore different sets of metrics. With a more focused set of plots, the workspace loads faster. Metrics that are not plotted are still collected and stored as usual.

Reviewer: Should the initial sentence be something like "When logging more than 5000 metrics, ...", then in a following sentence state the maximum we recommend is 100,000 metrics? (What's the recommendation above 100,000? Set up a separate project?

Author: I have no idea. I have had a JIra ticket to try to get latest performance info  (or at least some source of truth) for months now...

1. The size of your data is too large. Large data sizes could introduce a >1 ms overhead to the training loop.
2. The speed of your network and how the W&B backend is configured

> 3. Calling `run.log` more than a few times per second. This is due to a small latency added to the training loop every time `run.log` is called.

Reviewer: The first sentence is a fragment.

### Rate limits on metric logging API

> The `run.log` calls in your script utilize a metrics logging API to log your training data to W&B. This API is engaged through either online or [offline syncing]({{< relref "/ref/cli/wandb-sync.md" >}}). In either case, it imposes a rate limit quota limit in a rolling time window. This includes limits on total request size and request rate, where latter refers to the number of requests in a time duration.

Reviewer: This paragraph could use some love. "a metrics logging API" is really "the", right? 
```suggestion
When your script calls `run.log`, the metrics logging API logs your training data to W&B. The metrics logging API can run in online or [offline syncing]({{< relref "/ref/cli/wandb-sync.md" >}}) mode. In both modes, a rate limit quota limit is enforced in a rolling time window. The quota limits both total request size and the rate of requests over time.
```

```python
if epoch % 5 == 0:  # Log metrics every 5 epochs
    run.log({"acc": accuracy, "loss": loss})
```

- Manual data syncing: W&B store your run data locally if you are rate limited. You can manually sync your data with the command `wandb sync <run-file-path>`. For more details, see the [`wandb sync`]({{< relref "/ref/cli/wandb-sync.md" >}}) reference.

### Rate limits on GraphQL API

> The W&B Models UI and SDK’s [public API]({{< relref "/ref/python/public-api/api" >}}) make GraphQL requests to the server for querying and modifying data. For all GraphQL requests in SaaS Cloud, W&B applies rate limits per IP address for unauthorized requests and per user for authorized requests. The limit is based on request rate (request per second) within a fixed time window, where your pricing plan determines the default limits. For relevant SDK requests that specify a project path (for example, reports, runs, artifacts), W&B applies rate limits per project, measured by database query time.

Reviewer: ```suggestion
The W&B Models App and the SDK’s [public API]({{< relref "/ref/python/public-api/api" >}}) make GraphQL requests to the server for querying and modifying data. For all GraphQL requests in SaaS Cloud, W&B applies rate limits per IP address for unauthorized requests and per user for authorized requests. The limit is based on request rate (request per second) within a fixed time window, where your pricing plan determines the default limits. For relevant SDK requests that specify a project path (for example, `/reports`, '/runs`, or `/artifacts`), W&B applies rate limits to the entire project, measured by database query time.
```

#### Suggestions for staying under the GraphQL API rate limit

> If you are fetching a large volume of data using the W&B Models SDK's [public API]({{< relref "/ref/python/public-api/api" >}}), consider waiting at least one second between requests. If you receive a HTTP `429` `Rate limit exceeded` error or see `RateLimit-Remaining=0` in the response headers, wait for the number of seconds specified in `RateLimit-Reset` before retrying.

Reviewer: Do we like "W&B Models SDK's public API"? It seems a bit wordy, and similar content is in several places.
